### PR TITLE
fix(automate): update empty state

### PR DESCRIPTION
**Pull request by cdriesler**
## Description & motivation

- The empty state for the automations tab is functional but unhelpful
- We have a neat tool for this already being used in projects

## Changes:

- Borrows the project onboarding cards for first-use automate actions
- Blocks certain actions where relevant (i.e. disable automation creation if you don't have any workspace functions)

From:

Screen Shot 2024-12-04 at 18 29 02

To:

Screen Shot 2024-12-05 at 12 13 13

Copy to be iterated on, but these buttons will be important for funnel metrics as well.

### CODE REVIEW

File: packages/frontend-2/components/project/page/automations/EmptyState.vue [hunk 1]
```diff
@@ -1,48 +1,130 @@
...
+  }>(),
+  {
+    hiddenActions: () => [],
+    disabledActions: () => []
+  }
+)
+
+const router = useRouter()
 
 const functionsGalleryRoute = computed(() =>
```
Comment (andrewwallacespeckle) [inline packages/frontend-2/components/project/page/automations/EmptyState.vue:52]: This is showing as unused

File: packages/frontend-2/components/common/Card.vue [hunk 2]
```diff
@@ -21,20 +21,25 @@
...
         :key="button.id || index"
-        v-bind="button.props || {}"
-        :disabled="button.props?.disabled || button.disabled"
-        :submit="button.props?.submit || button.submit"
-        target="_blank"
-        external
-        size="sm"
-        color="outline"
-        @click="($event) => button.onClick?.($event)"
+        v-tippy="button.disabledMessage"
```
Comment (andrewwallacespeckle) [inline packages/frontend-2/components/common/Card.vue:27]: I see you have mastered the ways of v-tippy

**Review by andrewwallacespeckle — CHANGES_REQUESTED** [review]

File: packages/frontend-2/components/project/page/automations/EmptyState.vue (hunk 1 above)
Comment (cdriesler) [reply packages/frontend-2/components/project/page/automations/EmptyState.vue:52]: It is (should be) used on line 93 for `router.push(functionsGalleryRoute.value)`. Maybe a better way to do conditional routes like this?

File: packages/frontend-2/components/common/Card.vue (hunk 2 above)
Comment (cdriesler) [reply packages/frontend-2/components/common/Card.vue:27]: praise be

File: packages/frontend-2/components/project/page/automations/EmptyState.vue (hunk 1 above)
Comment (andrewwallacespeckle) [reply packages/frontend-2/components/project/page/automations/EmptyState.vue:52]: No, it was just flagged as unused in my IDE. But you're right, it is used. Oddly the warning has gone now too. Feel free to ignore, not sure what happened

**Pull request merged**